packages/core/src/modules/connections/services/helpers.ts

```diff
@@ -0,0 +1,49 @@
+import type { DidDocument } from '../../dids'
+import type { DidDoc, PublicKey } from '../models'
+
+import { KeyType } from '../../../crypto'
+import { AriesFrameworkError } from '../../../error'
+import { DidPeer, Key, DidDocumentBuilder } from '../../dids'
+import { getEd25519VerificationMethod } from '../../dids/domain/key-type/ed25519'
+import { PeerDidNumAlgo } from '../../dids/methods/peer/DidPeer'
+
+export function convertToNewDidDocument(didDoc: DidDoc): DidDocument {
+  const didDocumentBuilder = new DidDocumentBuilder('')
+
+  didDoc.authentication.forEach((auth) => {
+    const { publicKey: pk } = auth
+    if (pk.type === 'Ed25519VerificationKey2018' && pk.value) {
+      const ed25519VerificationMethod = convertPublicKeyToVerificationMethod(pk)
+      didDocumentBuilder.addAuthentication(ed25519VerificationMethod)
+    }
+  })
+
+  didDoc.publicKey.forEach((pk) => {
+    if (pk.type === 'Ed25519VerificationKey2018' && pk.value) {
+      const ed25519VerificationMethod = convertPublicKeyToVerificationMethod(pk)
+      didDocumentBuilder.addVerificationMethod(ed25519VerificationMethod)
+    }
+  })
```

Comment:
How does this work with references?

Reply:
It doesn't. How can I create `DidDoc` with referenced auth? if I do the following it doesn't work:
```js
const didDoc = new DidDoc({
  authentication: [
    new EmbeddedAuthentication(
      new Ed25119Sig2018({
        id: 'did:sov:SKJVx2kn373FNgvff1SbJo#4',
        controller: 'did:sov:SKJVx2kn373FNgvff1SbJo',
        publicKeyBase58: 'EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d',
      })
    ),
    new ReferencedAuthentication(
      new Ed25119Sig2018({
        id: 'did:sov:SKJVx2kn373FNgvff1SbJo#5',
        controller: 'did:sov:SKJVx2kn373FNgvff1SbJo',
        publicKeyBase58: 'EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d',
      }),
      'Ed25519VerificationKey2018'
    ),
  ],
  id: 'test-id',
  publicKey: [
    new Ed25119Sig2018({
      id: 'did:sov:SKJVx2kn373FNgvff1SbJo#4',
      controller: 'did:sov:SKJVx2kn373FNgvff1SbJo',
      publicKeyBase58: 'EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d',
    }),
    new Ed25119Sig2018({
      id: 'did:sov:SKJVx2kn373FNgvff1SbJo#5',
      controller: 'did:sov:SKJVx2kn373FNgvff1SbJo',
      publicKeyBase58: 'EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d',
    }),
  ],
  service: [
    new IndyAgentService({
      id: 'did:sov:SKJVx2kn373FNgvff1SbJo#service-1',
      serviceEndpoint: 'did:sov:SKJVx2kn373FNgvff1SbJo',
      recipientKeys: ['EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d'],
      routingKeys: ['EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d'],
      priority: 5,
    }),
  ],
})
```

I would expect that it returns this JSON:
```js
{
  "context": "https://w3id.org/did/v1",
  "publicKey": [
    {
      "id": "did:sov:SKJVx2kn373FNgvff1SbJo#4",
      "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
      "type": "Ed25519VerificationKey2018",
      "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
    },
    {
      "id": "did:sov:SKJVx2kn373FNgvff1SbJo#5",
      "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
      "type": "Ed25519VerificationKey2018",
      "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
    }
  ],
  "service": [
    {
      "id": "did:sov:SKJVx2kn373FNgvff1SbJo#service-1",
      "serviceEndpoint": "did:sov:SKJVx2kn373FNgvff1SbJo",
      "type": "IndyAgent",
      "priority": 5,
      "recipientKeys": ["EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"],
      "routingKeys": ["EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"]
    }
  ],
  "authentication": [
    {
      "publicKey": {
        "id": "did:sov:SKJVx2kn373FNgvff1SbJo#4",
        "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
        "type": "Ed25519VerificationKey2018",
        "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
      },
      "type": "Ed25519VerificationKey2018"
    },
    "did:sov:SKJVx2kn373FNgvff1SbJo#5"
  ],
  "id": "test-id"
}

```

But it returns this one
```
{
  "context": "https://w3id.org/did/v1",
  "publicKey": [
    {
      "id": "3",
      "controller": "did:sov:LjgpST2rjsoxYegQDRm7EL",
      "type": "RsaVerificationKey2018",
      "value": "-----BEGIN PUBLIC X..."
    },
    {
      "id": "did:sov:SKJVx2kn373FNgvff1SbJo#4",
      "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
      "type": "Ed25519VerificationKey2018",
      "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
    },
    {
      "id": "did:sov:SKJVx2kn373FNgvff1SbJo#5",
      "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
      "type": "Ed25519VerificationKey2018",
      "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
    },
    {
      "id": "6",
      "controller": "did:sov:LjgpST2rjsoxYegQDRm7EL",
      "type": "Secp256k1VerificationKey2018",
      "value": "-----BEGIN PUBLIC A..."
    }
  ],
  "service": [
    {
      "id": "did:sov:SKJVx2kn373FNgvff1SbJo#service-1",
      "serviceEndpoint": "did:sov:SKJVx2kn373FNgvff1SbJo",
      "type": "IndyAgent",
      "priority": 5,
      "recipientKeys": ["EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"],
      "routingKeys": ["EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"]
    }
  ],
  "authentication": [
    {
      "publicKey": {
        "id": "did:sov:SKJVx2kn373FNgvff1SbJo#4",
        "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
        "type": "Ed25519VerificationKey2018",
        "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
      }
    },
    {
      "publicKey": {
        "id": "did:sov:SKJVx2kn373FNgvff1SbJo#5",
        "controller": "did:sov:SKJVx2kn373FNgvff1SbJo",
        "type": "Ed25519VerificationKey2018",
        "value": "EoGusetSxDJktp493VCyh981nUnzMamTRjvBaHZAy68d"
      },
      "type": "Ed25519VerificationKey2018"
    }
  ],
  "id": "test-id"
}
```